libs/src/sdk/services/DiscoveryNodeSelector/DiscoveryNodeSelector.ts

```diff
@@ -246,119 +240,138 @@ export class DiscoveryNodeSelector implements DiscoveryNodeSelectorService {
    * @returns a healthy discovery node endpoint
    */
   private async select() {
-    this.debug('Selecting new discovery node...')
-    const decisionTree: Decision[] = []
-
-    // Get all the services we have
-    let services = [...this.services]
-    decisionTree.push({
-      stage: DECISION_TREE_STATE.GET_ALL_SERVICES,
-      val: services
-    })
-
-    // If a whitelist is provided, filter down to it
-    if (this.config.allowlist) {
-      services = services.filter((s) => this.config.allowlist?.has(s))
-      decisionTree.push({
-        stage: DECISION_TREE_STATE.FILTER_TO_WHITELIST,
-        val: services
+    if (this.reselectLock) {
```

Comment:
i prefer it here because `this.select` is often called on its own outside of `getSelectedEndpoint` and ultimately we want to avoid two+ ongoing reselections at once

Reply:
oof, you're right! good call...